slack.php

```diff
@@ -57,90 +69,115 @@
-        unset($urlParams['text']);
-    }
+        $server = \Deployer\Task\Context::get()->getServer();
+        if ($server instanceof \Deployer\Server\Local) {
+            $user = get('local_user');
+        } else {
+            $user = $server->getConfiguration()->getUser() ?: null;
+        }
 
-    foreach (['parse', 'link_names', 'icon_url', 'unfurl_links', 'unfurl_media', 'as_user'] as $option) {
-        if (isset($config[$option])) {
-            $urlParams[$option] = $config[$option];
+        $messagePlaceHolders = [
+            '{{release_path}}' => get('release_path'),
+            '{{host}}' => get('server.host'),
+            '{{stage}}' => get('stages')[0],
+            '{{user}}' => $user,
+            '{{branch}}' => get('branch'),
+            '{{app_name}}' => isset($config['app']) ? $config['app'] : 'app-name',
+            '{{commits}}' => $commits
+        ];
+        $config['message'] = strtr($config['message'], $messagePlaceHolders);
+
+        $urlParams = [
+            'channel' => $config['channel'],
+            'token' => $config['token'],
+            'text' => $config['message'],
+            'username' => $config['username'],
+            'icon_emoji' => $config['icon'],
+            'pretty' => true
+        ];
+
+        if ($config['unset_text']) {
+            unset($urlParams['text']);
         }
-    }
 
-    if (isset($config['attachments'])) {
-        $urlParams['attachments'] = json_encode($config['attachments']);
-    }
+        if ($config['use_text_in_attachment']) {
+            $config['attachments'][0]['text'] = $config['message'];
+        }
 
-    if (isset($config['icon_url'])) {
-        unset($urlParams['icon_emoji']);
-    }
+        foreach (['parse', 'link_names', 'icon_url', 'unfurl_links', 'unfurl_media', 'as_user'] as $option) {
+            if (isset($config[$option])) {
+                $urlParams[$option] = $config[$option];
+            }
+        }
 
-    $url = 'https://slack.com/api/chat.postMessage?' . http_build_query($urlParams);
-    $result = @file_get_contents($url);
+        if (isset($config['attachments'])) {
+            $urlParams['attachments'] = json_encode($config['attachments']);
+        }
 
-    if (!$result) {
-        throw new \RuntimeException($php_errormsg);
-    }
+        if (isset($config['icon_url'])) {
+            unset($urlParams['icon_emoji']);
+        }
+
+        $url = 'https://slack.com/api/chat.postMessage?' . http_build_query($urlParams);
+        $result = @file_get_contents($url);
+
+        if (!$result) {
+            throw new \RuntimeException($php_errormsg);
+        }
 
-    $response = @json_decode($result);
+        $response = @json_decode($result);
 
-    if (!$response || isset($response->error)) {
-        throw new \RuntimeException($response->error);
+        if (!$response || isset($response->error)) {
+            throw new \RuntimeException($response->error);
+        }
     }
+    exit;
```

Comment:
Sorry, that was for testing because I was missing a --no-hook parameter (see https://github.com/deployphp/deployer/pull/934)